utils/compare/compute-comparators.ts

```diff
@@ -0,0 +1,22 @@
+import type {
+  ComparatorByOptionsComputer,
+  Comparator,
+} from './default-comparator-by-options-computer'
+import type { CommonOptions } from '../../types/common-options'
+import type { SortingNode } from '../../types/sorting-node'
+
+export function computeComparators<
+  Options extends CommonOptions,
+  T extends SortingNode,
+>(
+  comparatorByOptionsComputer: ComparatorByOptionsComputer<Options, T>,
+  options: Options,
+): Comparator<T>[] {
+  return [
+    comparatorByOptionsComputer(options),
```

Comment:
With type: `'unsorted'` we still add the fallback comparator, so the array ends up sorted, whereas the previous compare short-circuited and preserved order. Is this behavior intentional? 

If not, maybe skip adding the fallback comparator when `type === 'unsorted'` so such groups remain unsorted.